stdlib/public/core/Codable.swift

```diff
@@ -0,0 +1,3176 @@
+    public func decodeIfPresent(_ type: UInt64.Type, forKey key: Key) throws -> UInt64? {
+        return try box.decodeIfPresent(UInt64.self, forKey: key)
+    }
+
+    /// Decodes a value of the given type for the given key, if present.
+    ///
+    /// This method returns `nil` if the container does not have a value associated with `key`, or if the value is null. The difference between these states can be distinguished with a `contains(_:)` call.
+    ///
+    /// - parameter type: The type of value to decode.
+    /// - parameter key: The key that the decoded value is associated with.
+    /// - returns: A decoded value of the requested type, or `nil` if the `Decoder` does not have an entry associated with the given key, or if the value is a null value.
+    /// - throws: `DecodingError.typeMismatch` if the encountered encoded value is not convertible to the requested type.
+    public func decodeIfPresent(_ type: Float.Type, forKey key: Key) throws -> Float? {
+        return try box.decodeIfPresent(Float.self, forKey: key)
+    }
+
+    /// Decodes a value of the given type for the given key, if present.
+    ///
+    /// This method returns `nil` if the container does not have a value associated with `key`, or if the value is null. The difference between these states can be distinguished with a `contains(_:)` call.
+    ///
+    /// - parameter type: The type of value to decode.
+    /// - parameter key: The key that the decoded value is associated with.
+    /// - returns: A decoded value of the requested type, or `nil` if the `Decoder` does not have an entry associated with the given key, or if the value is a null value.
+    /// - throws: `DecodingError.typeMismatch` if the encountered encoded value is not convertible to the requested type.
+    public func decodeIfPresent(_ type: Double.Type, forKey key: Key) throws -> Double? {
+        return try box.decodeIfPresent(Double.self, forKey: key)
+    }
+
+    /// Decodes a value of the given type for the given key, if present.
+    ///
+    /// This method returns `nil` if the container does not have a value associated with `key`, or if the value is null. The difference between these states can be distinguished with a `contains(_:)` call.
+    ///
+    /// - parameter type: The type of value to decode.
+    /// - parameter key: The key that the decoded value is associated with.
+    /// - returns: A decoded value of the requested type, or `nil` if the `Decoder` does not have an entry associated with the given key, or if the value is a null value.
+    /// - throws: `DecodingError.typeMismatch` if the encountered encoded value is not convertible to the requested type.
+    public func decodeIfPresent(_ type: String.Type, forKey key: Key) throws -> String? {
+        return try box.decodeIfPresent(String.self, forKey: key)
+    }
+
+    /// Decodes a value of the given type for the given key, if present.
+    ///
+    /// This method retu
```

Comment:
Certainly, I could not think of any reason to make `decode()` of another protocol `mutating` as well.
Thank you for answering my boring questions. :pray: